src/main/java/org/ecocean/StartupWildbook.java

```diff
@@ -290,19 +290,18 @@ private static void startWildbookScheduledTaskThread(String context) {
                 System.out.println("[INFO]: checking for scheduled tasks to execute...");
                 Shepherd myShepherd = new Shepherd(context);
                 myShepherd.setAction("WildbookScheduledTaskThread");
+                myShepherd.beginDBTransaction();
                 try {
-                    ArrayList<WildbookScheduledTask> scheduledTasks =
-                    myShepherd.getAllIncompleteWildbookScheduledTasks();
+                    ArrayList<WildbookScheduledTask> scheduledTasks = myShepherd.getAllIncompleteWildbookScheduledTasks();
                     for (WildbookScheduledTask scheduledTask : scheduledTasks) {
                         if (scheduledTask.isTaskEligibleForExecution()) {
                             scheduledTask.execute(myShepherd);
                         }
                     }
                 } catch (Exception e) {
-                    myShepherd.rollbackAndClose();
```

Comment:
i am wondering if this `rollbackAndClose()` might cause something in the task execution to not be persisted, for example, there is a `myShepherd.throwAwayMarkedIndividual()` buried in `mergeIndividuals()` that i _think_ might be affected by this. will expand a little on this in slack....

Reply:
Hi @naknomum 

The WildbookScheduledTask.execute() method (and anything inheriting from it) is responsible for its own persistence with this Shepherd and for handling its exceptions in a way that allows other tasks to still execute. I think this method is doing the right thing by serving as an overall catch-all, but I did update ScheduledIndividualMerge to ensure it unwound its changes on exception.